ironfish-cli/STYLE_GUIDE.md

```diff
@@ -0,0 +1,68 @@
+
+Based on https://devcenter.heroku.com/articles/cli-style-guide
+
+## Naming the command
+The CLi is made up of topics and commands. For the command ironfish migrations:start, migrations is the topic and start is the command.
+
+Generally topics are plural nouns and commands are verbs.
+
+Topic and command names should always be a single, lowercase word without spaces, hyphens, underscores, or other word delimiters. Colons, however, are allowed as this is how to define subcommands (such as `ironfish wallet:transactions:add`). If there is no obvious way to avoid having multiple words, separate with kebab-case: `ironfish service:estimate-fee-rates`
+
+Because topics are generally nouns, the root command of a topic usually lists those nouns. So in the case of `ironfish migrations`, it will list all the migrations. Never create a *:list command such as `ironfish migrations:list`.
+
+## Command Description
+Topic and command descriptions should be provided for all topics and commands. They should fit on 80 character width screens, begin with a lowercase character, and should not end in a period.
+
+## Input
+Input to commands is typically provided by flags and args. Stdin can also be used in cases where it is useful to stream files or information in (such as with heroku run).
+
+### Flags
+
+Flags are preferred to args when there are many inputs, particularly inputs of the same type. They involve a bit more typing, but make the use of the CLI clearer. For example, `ironfish wallet:send` used to accept an argument for the account to use, as well as --to flag to specify the account to send to.
+
+So using heroku fork used to work like this:
+
+```bash
+$ ironfish wallet:send source_account --to dest_account
+```
+
+This is confusing to the user since it isn’t clear which account they are sending from and which one they are sending to. By switching to required flags, we instead expect input in this form:
+
+```bash
+ironfish wallet:send --account source_account --to dest_account
+```
+
+This also allows the user to specify the flags in any order, and gives them the confidence that they are running the command correctly. It also allows us to show better error messages.
+
+Ensure that descriptions are provided for all flags, that the descriptions are in lowercase, that they are concise (so as to fit on narrow screens), and that they do not end in a period to match other flag descriptions.
+
+
+### Arguments
+Arguments are the basic way to provide input for a command. While flags are generally preferred, they are sometimes unnecessary in cases where there is only 1 argument, or the arguments are obvious and in an obvious order.
+
+In the case of `ironfish chain:power`, we can specify the account we want to give access to with an argument.
```

Comment:
"we can specify the account we want to give access to with an argument." - this doesnt match what the chain:power command does, guessing this used to be a different command?

Reply:
Bad description, oops. I changed the example to chain:power from wallet:use but didnt update the text.

Reply:
Fixed